parser.y

```diff
@@ -9175,42 +9176,71 @@ SelectLockOpt:
 	{
 		$$ = nil
 	}
-|	"FOR" "UPDATE"
+|	"FOR" "UPDATE" OfTablesOpt
 	{
-		$$ = &ast.SelectLockInfo{LockType: ast.SelectLockForUpdate}
+		$$ = &ast.SelectLockInfo{
+			LockType: ast.SelectLockForUpdate,
+			Tables:   $3.([]*ast.TableName),
+		}
 	}
-|	"FOR" "SHARE"
+|	"FOR" "SHARE" OfTablesOpt
```

Comment:
We don't need to support `FOR SHARE`.

Reply:
I'm referring to the postgres syntax https://www.postgresql.org/docs/current/sql-select.html. `for share` would have been invalid as well, but the syntax seems to be fine for full support.